nomad/mock/mock.go

```diff
@@ -2331,3 +2338,48 @@ func SecureVariable() *structs.SecureVariable {
 	}
 	return sv
 }
+
+// SecureVariables returns a random number of secure variables between min
+// and max inclusive.
+func SecureVariables(minU, maxU uint8) MockSecureVariables {
```

Comment:
Does randomness here get us a lot for unit testing? If we do run into a case where a test fails randomly because of this, it feels like it's tricky to reproduce the test case that caused the problem.

Reply:
Fair point.  I guess the thing that I appreciate about randomized mocking is that I didn't have to add extra testing code to make the values different enough to ensure that I'm not getting false successes. This mock generator could also be used to make hundreds of values for testing pagination and the UI at scale. Perhaps there is use for both approaches—make some static mocks for units and then have the random one for fuzz testing the units or just generating a large amount of faux data? I will defer to your call on this one; just wanted to add the extra context.